src/components/Details/Details.js

```diff
@@ -14,36 +16,29 @@ class Details extends React.Component {
         this.state = {
             errorMessage: '',
             movie: {},
-            movieTrailers: [],
+            movieKey: '',
         };
     }
 
     componentDidMount() {
 
-        let getMovie = (endpoint) => {
-           let fetchData =  
-           endpoint ?
-           fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies/${this.props.id}/${endpoint}`) 
-           : fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies/${this.props.id}`);
-
-        return fetchData
-                .then(response => response.json())
-                .then(data => {
-                if(data.movie) {
-                let movie = data.movie
-                this.setState({movie})
-                } else if(data.videos.length) {
-                    this.setState({movieTrailers: data.videos})
-                }
+        getMovieDetails(this.props.id)
+            .then(data => {
+                const movieDetails = filterMovieDetails(data)
+                this.setState({movie: movieDetails})
             })
             .catch(err => {
-                console.error(err, 'Error caught in get request for movie details')
                 this.setState({errorMessage: 'Something went wrong, please try again later 😔'})
             });
-            }
 
-          Promise.all([getMovie(), getMovie('videos')])
-          .then(promise => console.log(promise));
+        getMovieTrailer(this.props.id)
```

Comment:
This is very clean and easy to read, my only concern is that without a promise wrapping everything, we will render a movie details page or the youtube video before the rest of the details.